### PR TITLE
Issue#27 : HA implementation with zk/curator leader election recipe.

### PR DESCRIPTION
**Pull request by satishd**
This PR addresses issue #27 
 - HA implementation with zk/curator leader election recipe.
 - Added HA sanity test by using local registry servers

### CODE REVIEW

**Comment by satishd** [comment]
ping @harshach

File: common/src/main/java/com/hortonworks/registries/common/ha/LeadershipClient.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+     * or it has already been closed.
+     *
+     * @throws IOException if any IO related errors occur.
+     */
+    void exitFromLeaderParticipation() throws IOException;
+
+    /**
+     * Returns true if the current participant is a leader
+     */
+    boolean hasLeadership();
```
Comment (hmcl) [inline common/src/main/java/com/hortonworks/registries/common/ha/LeadershipClient.java]: isLeader

File: common/src/main/java/com/hortonworks/registries/common/ha/LeadershipClient.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ */
+package com.hortonworks.registries.common.ha;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Client to connect to target systems(like ZooKeeper) which give a way to elect a leader among the participants.
+ */
+public interface LeadershipClient {
```
Comment (hmcl) [inline common/src/main/java/com/hortonworks/registries/common/ha/LeadershipClient.java]: LeaderElection? LeaderElectionClient? LeaderElector? Leadership is not a name very commonly used

Comment (satishd) [inline common/src/main/java/com/hortonworks/registries/common/ha/LeadershipClient.java]: My preference would be `LeadershipParticipant`, avoiding election in the name.

**Comment by harshach** [comment]
+1

**Pull request merged**